source/common/router/config_impl.cc

```diff
@@ -923,6 +929,28 @@ RouteConstSharedPtr RegexRouteEntryImpl::matches(const Http::RequestHeaderMap& h
   return nullptr;
 }
 
+ConnectRouteEntryImpl::ConnectRouteEntryImpl(
+    const VirtualHostImpl& vhost, const envoy::config::route::v3::Route& route,
+    Server::Configuration::ServerFactoryContext& factory_context,
+    ProtobufMessage::ValidationVisitor& validator)
+    : RouteEntryImplBase(vhost, route, factory_context, validator) {}
+
+void ConnectRouteEntryImpl::rewritePathHeader(Http::RequestHeaderMap& headers,
+                                              bool insert_envoy_original_path) const {
+  const absl::string_view path =
+      Http::PathUtil::removeQueryAndFragment(headers.Path()->value().getStringView());
```

Comment:
We need to land the other PR for this to be safe, right? As an aside, any thoughts on making it programatically impossible to mess this up? Now that we split the header map APIs, should we special case Path so it returns an optional or something like that? WDYT? cc @asraa 

It might also be good to split the config_impl changes/tests out from the rest of this PR, but up to you.

Reply:
We don't, because the HCM still adds the synthetic "/" path header.

Reply:
Regarding impossible to mess up (which I could do with a follow-up after this and before removing the synthetic path header, I think if we wanted to do that we might want to do it with a bunch of the "expected but not guaranteed" headers.  My preference would be to have a filter fuzzer / filter compliance suite where every filter is hit with pathless requests, but also other common unusual scenarios like hostless responses.

Reply:
Yes this would be great, agreed. I think we are going to have a very common class of bugs where filters thing that certain headers will always be set which we are now breaking.